### ISSUE TITLE
Your multi-row tab solution for Firefix is miles behind Quantum Nox Dark Theme's solution

### ISSUE THREAD

**Issue by smihaila** [MrOtherGuy/firefox-csshacks#240]
I'm sorry, but I personally consider your "solution" to multi-row tabs for Firefox, a piece of shit. Mainly for 3 reasons:

1. The tabs cannot be reordered.
2. It tampers with the main window's title bar and minimize / maximize / close system control icons that are shown in the top-right corner of the main window.
3. It allows only 3 rows at a time to be visible, and displaying a shitty "scroll bar" which you can't even click with the mouse, to scroll.

Quantum Nox Dark Theme's solution is so much superior, and in so many ways. It's too bad that it no longer works with Firefox versions beyond 104.x, for example the 107.x or 108.x.

If you could make it work like Quantum Nox's, then you'll get a fan for life.
I'm currently keeping Firefox updates in disabled state, mainly because I can't find a good solution for Firefox post-104.0

You currently solution well ... simply sucks, big time.

**Comment by mcdonnnj** [comment]
I mean put bluntly if you don't like a project then do one of the following:

1. Leave constructive feedback.
2. Contribute to improve the project.
3. Don't use it and move on.

You've managed to do none of these things. In this particular case if you are such a fan of Quantum Nox you can fork the archived repository and update it to work with newer versions of Firefox instead of leaving rude issues in other users' projects.

**Comment by MrOtherGuy** [comment]
I know it doesn't work 100% well, and I think it says that in the file. Feel free to open a pull request if you have some improvements to it though.

But as far I know, Qunatum Nox is not a CSS mod, at least not completely. Instead injects custom javascript logic to Firefox that changes whole bunch of things, and as such it can change how tab handling in multi-row mode works.

This repository is for CSS hacks, for things you can achieve by simply changing styling rules.

**Comment by sleek22** [comment]
Just to be clear, I'm answering the third point for the benefit of others who might be reading this, not for the thankless person posting here.

_It allows only 3 rows at a time to be visible, and displaying a shitty "scroll bar" which you can't even click with the mouse, to scroll._

For the second line in the code, I changed to this to get 20 rows:

--multirow-n-rows: 20;

Also, to MrOtherGuy, I can't thank you enough for the work you do in making Firefox more useable for us. I'm sure I speak for many others when I say your work is truly appreciated.

Thank you again.

**Comment by smihaila** [comment]
Sorry for being too abrasive about this, but the whole multi-row tab support situation (rather the lack thereof) is simply frustrating.

We are at the end of 2022, and Mozilla is blatantly adverse to multi-row tabs. Soon following Google's arrogance (whose so-called "system architects" and "UX designers" are "telling us" - from a high horse position, what our needs _should _be__ instead of listening to what the Users' **wants** really are - I would slap their hands with a keyboard, if I could).

I've found out in the meantime that TabMixPlus can work with the latest Firefox 108.0 release, via the "userChromeJS" layer, which brings back "legacy" support, thus allowing the TabMixPlus' .xpi modules to be loaded:
https://github.com/onemen/TabMixPlus
https://github.com/xiaoxiaoflood/firefox-scripts#instructions

I've just tried it, and it seems to be working very well. So it looks like I've found a great replacement solution to Quantum Nox, and working with FF 108.0

[I was also trying - on a 2nd PC - another solution which is quite similar to your firefox-csshacks, called PAXMOD, and which necessitated installing a Firefox Developer edition. But since I've found my best solution, I'll no longer need FF Developer edition installed.]

Anyways, best of luck with your project!
